pkg/miniogw/gateway-storj.go

```diff
@@ -72,48 +242,106 @@ func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
 }
 
 func (s *storjObjects) GetObjectInfo(ctx context.Context, bucket,
-	object string) (objInfo cmd.ObjectInfo, err error) {
+	object string) (objInfo minio.ObjectInfo, err error) {
 	panic("TODO")
 }
 
 func (s *storjObjects) ListBuckets(ctx context.Context) (
-	buckets []cmd.BucketInfo, err error) {
-	return []cmd.BucketInfo{{
-		Name:    "test-bucket",
-		Created: time.Now(),
-	}}, nil
+	buckets []minio.BucketInfo, err error) {
+	buckets, err = getbucketscallback()
```

Comment:
so here, you should be able to call s.s.getbucketscallback() instead, or just put all that logic in here. that way it's defined on a struct and you don't need global memory

Reply:
Modified as suggested to eliminate global references